### DISCUSSION TITLE
custom output consumer?

### DISCUSSION THREAD

**Discussion by GustavoKatel** [nvim-neotest/neotest#46] [Q&A] [answered]
hey thanks for putting this together, I was looking for something like this.

I was wondering if it would make sense to allow consumers to be defined as plugins, similar to adapters. So people can create custom consumers (such as output window or other integrations) without the need to change the core.

thoughts?

**Comment by rcarriga** [comment, marked as answer]
I think it would absolutely make sense! I just haven't gotten to implementing it.

It'll probably be something like this
```lua
consumers = {
  plugin_consumer = require("neotest-plugin-consumer"),
  personal_consumer = function(client)
    return {
      some_func =  function ()
        ...
      end,
    }
  end,
}
```
and so then users can call `require("neotest").personal_consumer.some_func` like any other consumer.

**Comment by rcarriga** [comment]
I've added the ability to specify custom consumers, using the same config as above.

**Reply by GustavoKatel** [reply]
this is great! thanks!
